### PR TITLE
Refactor GPU example containers

### PR DESCRIPTION
**Pull request by mengxr**
* Use NVIDIA official CUDA cuDNN runtime as the base.
* Make a dedicated conda layer without root conda environment to simplify the build of TF, Torch, and RAPIDS.
* Some clean up to reduce the image sizes (RAPIDS 8.2GB -> 5.2GB)
* Include `build.sh` to help rebuild all images.

All images were published to my docker hub: https://hub.docker.com/u/mengxr

### CODE REVIEW

File: ubuntu/gpu/base/Dockerfile [hunk 1]
```diff
@@ -1,57 +1,20 @@
-FROM databricksruntime/minimal:latest
+# See https://gitlab.com/nvidia/container-images/cuda/-/blob/master/dist/ubuntu18.04/10.1/runtime/cudnn7/Dockerfile
+FROM nvidia/cuda:10.1-cudnn7-runtime-ubuntu18.04
```
Comment (evanye) [inline ubuntu/gpu/base/Dockerfile:2]: What is the motivation for switching the base image?

File: ubuntu/gpu/rapids/Dockerfile [hunk 2]
```diff
@@ -1,9 +1,9 @@
-FROM databricksruntime/gpu-python:cuda10.1
+FROM databricksruntime/gpu-conda:cuda10.1
```
Comment (evanye) [inline ubuntu/gpu/rapids/Dockerfile:1]: don't understand how this reduces the image size so much

File: ubuntu/gpu/base/Dockerfile (hunk 1 above)
Comment (mengxr) [reply ubuntu/gpu/base/Dockerfile:2]: The NVIDIA dependencies are tricky to manage. For example, cuBLAS 10.2.2 is not compatible with CUDA 10.1 while cuBLAS 10.2.1 is compatible ... it is very easy to make mistakes. So the safest approach is to use NVIDIA CUDA docker directly.

File: ubuntu/gpu/rapids/Dockerfile (hunk 2 above)
Comment (mengxr) [reply ubuntu/gpu/rapids/Dockerfile:1]: There are three major places we saved space in this version:
* NVIDIA cuda container does not install all the GPU libraries we pre-installed, e.g., the full CUDA toolkit and development libraries.
* We did a "conda clean --all" in the base environment.
* The RAPIDS container used to depend on gpu-python, which has a conda env "gpu" already.

**Pull request merged**